src/aocharmovie.cpp

```diff
@@ -61,6 +61,15 @@ void AOCharMovie::load_image(QString p_char, QString p_emote,
     return;
 
   m_reader->setFileName(emote_path);
+
+  // set format to apng if png supports animation
+  if (emote_path.endsWith("png")) {
+    m_reader->setFormat("apng");
+    if (!m_reader->supportsAnimation()) {
```

Comment:
This confused me at first because it first checks whether the reader itself supports animation, and *then* checks whether the image being read supports it as well. [source](https://code.woboq.org/qt5/qtbase/src/gui/image/qimagereader.cpp.html#_ZNK12QImageReader17supportsAnimationEv)